### PR TITLE
refactor: move networktime to world

### PR DESCRIPTION
**Pull request by uweenukr**
still works in host situations. another steps towards allowing multiple worlds per client/server.

### CODE REVIEW

**Comment by uweenukr** [comment]
pretty sure this will have a Doc error that ill work on later.

**Comment by uweenukr** [comment]
originally it was testing two different values. now both are essentially the same.

**Review by James-Frowen — CHANGES_REQUESTED** [review]

**Review by James-Frowen — APPROVED** [review]

**Comment by James-Frowen** [comment]
Nice change, just the one suggestion then it should be to merge

**Comment by James-Frowen** [comment]
Oh also make sure to add `BREAKING CHANGE: ...` to the commit when it gets merged

File: Assets/Mirage/Runtime/NetworkClient.cs [hunk 1]
```diff
@@ -137,7 +132,7 @@ public void Connect(string address = null, ushort? port = null)
             // setup all the handlers
             Player = new NetworkPlayer(connection);
             dataHandler.SetConnection(connection, Player);
-            Time.Reset();
+            World.Time.Reset();
```
[the remark this answers is not in the document]
Comment (uweenukr) [reply Assets/Mirage/Runtime/NetworkClient.cs:135]: there were actually a few similar optimizations I was going to include in another PR. but I can do them here.

File: Assets/Mirage/Runtime/NetworkBehaviour.cs [hunk 2]
```diff
@@ -119,7 +119,7 @@ public abstract class NetworkBehaviour : MonoBehaviour
         /// <summary>
         /// Returns the appropriate NetworkTime instance based on if this NetworkBehaviour is running as a Server or Client.
         /// </summary>
-        public NetworkTime NetworkTime => IsClient ? Client.Time : Server.Time;
+        public NetworkTime NetworkTime => World.Time;
```
Comment (Lymdun) [inline Assets/Mirage/Runtime/NetworkBehaviour.cs:122]: Much cleaner! Good job

File: Assets/Mirage/Runtime/NetworkClient.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (uweenukr) [reply Assets/Mirage/Runtime/NetworkClient.cs]: fixed

**Pull request merged**